lightning/common/pause.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"runtime"
+	"sync/atomic"
+)
+
+const (
```

Comment:
😄 why not use a channel to sequence operation, it's more simple

Reply:
See the explanation in https://github.com/pingcap/tidb-lightning/pull/199#discussion_r298452355 :)

Reply:
there is not much need for good performance here, but it doesn't matter, just enjoy this